R/rdeephaven/R/table_handle_wrapper.R

```diff
@@ -1,104 +1,123 @@
+  },
+  signature = c("table_handle_instance")
+)
 
-    #' @description
-    #' Binds the table referenced by this TableHandle to a variable on the server,
-    #' enabling it to be accessed by that name from any Deephaven API.
-    #' @param name Name for this table on the server.
-    bind_to_variable = function(name) {
-      .verify_string("name", name)
-      private$internal_table_handle$bind_to_variable(name)
-    },
+#' @export
+setMethod(
+  "is_static",
+  signature = c(table_handle_instance = "TableHandle"),
+  function(table_handle_instance) {
+    return(table_handle_instance@.internal_rcpp_object$is_static())
+  }
+)
 
-    #' @description
-    #' Imports the table referenced by this TableHandle into an Arrow RecordBatchStreamReader.
-    #' @return A RecordBatchStreamReader containing the data from the table referenced by this TableHandle.
-    to_arrow_record_batch_stream_reader = function() {
-      ptr <- private$internal_table_handle$get_arrow_array_stream_ptr()
-      rbsr <- RecordBatchStreamReader$import_from_c(ptr)
-      return(rbsr)
-    },
+#' @export
+setMethod(
+  "head",
+  signature = c(x = "TableHandle"),
+  function(x, n, ...) {
+    verify_positive_int("n", n, TRUE)
+    return(new("TableHandle", .internal_rcpp_object = x@.internal_rcpp_object$head(n)))
+  }
+)
 
-    #' @description
-    #' Imports the table referenced by this TableHandle into an Arrow Table.
-    #' @return A Table containing the data from the table referenced by this TableHandle.
-    to_arrow_table = function() {
-      rbsr <- self$to_arrow_record_batch_stream_reader()
-      arrow_tbl <- rbsr$read_table()
-      return(arrow_tbl)
-    },
+#' @export
+setMethod(
+  "tail",
+  signature = c(x = "TableHandle"),
+  function(x, n, ...) {
+    verify_positive_int("n", n, TRUE)
+    return(new("TableHandle", .internal_rcpp_object = x@.internal_rcpp_object$tail(n)))
+  }
+)
 
-    #' @description
-    #' Imports the table referenced by this TableHandle into a dplyr Tibble.
-    #' @return A Tibble containing the data from the table referenced by this TableHandle.
-    to_tibble = function() {
-      rbsr <- self$to_arrow_record_batch_stream_reader()
-      arrow_tbl <- rbsr$read_table()
-      return(as_tibble(arrow_tbl))
-    },
+#' @export
+setMethod(
+  "nrow",
+  signature = c(x = "TableHandle"),
+  function(x) {
+    return(x@.internal_rcpp_object$num_rows())
+  }
+)
 
-    #' @description
-    #' Imports the table referenced by this TableHandle into an R Data Frame.
-    #' @return A Data Frame containing the data from the table referenced by this TableHandle.
-    to_data_frame = function() {
-      arrow_tbl <- self$to_arrow_table()
-      return(as.data.frame(as.data.frame(arrow_tbl))) # TODO: for some reason as.data.frame on arrow table returns a tibble, not a data frame
-    }
-  ),
-  private = list(
-    internal_table_handle = NULL,
-    is_static_field = NULL
-  )
+# TODO: Implement ncol, dim
```

Comment:
todo

Reply:
At the moment I don't believe C++ can support this directly